src/app/modules/notification/services/notification.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+
+export interface Notification {
+  id: string;                   // auto generated unique id
+  message: string;              // html content
+  type: string;                 // 'default' | 'info' | 'warning' | 'success' => default: 'default'
+  options: NotificationOptions; // options
+}
+
+export interface NotificationOptions {
+  isDismissible: boolean;       // default: true
+  isSelfClosing: boolean;       // default: true
+  duration: number;             // delay in seconds before self closing => default: 5
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotificationService {
+
+  /** Notifications list */
+  private notifications: Notification[] = [];
+
+  /** Default notifiactions options */
+  private options: NotificationOptions = {
+    isDismissible: true,
+    isSelfClosing: true,
+    duration: 5
+  };
+
+  /** Id prefix of the nofications */
+  private idPrefix = 'n-';
```

Comment:
don't forget to change the prefix here if you rename the notif in the `html for` like I said before :)